src/host/stream.cpp

```diff
@@ -288,28 +288,36 @@ try
 {
     bytesRead = 0;
 
-    auto pending = readHandleState.GetPendingInput();
+    const auto pending = readHandleState.GetPendingInput();
+    auto input = pending;
 
+    // This is basically the continuation of COOKED_READ_DATA::_handlePostCharInputLoop.
     if (readHandleState.IsMultilineInput())
     {
-        const auto idx = pending.find(UNICODE_LINEFEED);
-        if (idx != decltype(pending)::npos)
-        {
-            // +1 to include the newline.
-            pending = pending.substr(0, idx + 1);
-        }
+        const auto firstLineEnd = input.find(UNICODE_LINEFEED) + 1;

```

Comment:
what happens if the linefeed is the end of the input and `+1` is off the end?

Reply:
In that case we crash because the function is `noexcept`. I'm personally fine with that, because more than just this code relies on our alias code to always properly terminate all lines in its output with exactly `\r\n`.

Reply:
Oh wait, this one is the non-`noexcept` one... But that's still fine IMO, because it got a `NT_CATCH_RETURN()` and it should never get hit (unless someone adds a major programming error into `alias.cpp`).

Reply:
Alright, now it can't happen anymore. 😄